examples/thermostat/silabs/src/SensorManager.cpp

```diff
@@ -116,11 +116,19 @@ void SensorManager::SensorTimerEventHandler(void * arg)
     if ((temperature >= (lastTemperature + kMinTemperatureDelta)) || temperature <= (lastTemperature - kMinTemperatureDelta))
     {
         lastTemperature = temperature;
-        PlatformMgr().LockChipStack();
+        AttributeUpdateInfo * data = chip::Platform::New<AttributeUpdateInfo>();
+        data->endPoint = kThermostatEndpoint;
+        data->temperature = temperature;
         // The SensorMagager shouldn't be aware of the Endpoint ID TODO Fix this.
         // TODO Per Spec we should also apply the Offset stored in the same cluster before saving the temp
 
-        app::Clusters::Thermostat::Attributes::LocalTemperature::Set(kThermostatEndpoint, temperature);
-        PlatformMgr().UnlockChipStack();
+        chip::DeviceLayer::PlatformMgr().ScheduleWork(UpdateClusterState, reinterpret_cast<intptr_t>(data));
     }
 }
+
+void SensorManager::UpdateClusterState(intptr_t context)
```

Comment:
The function name does not look correct.  UpdateTemperatureAttributes()?

Reply:
Modified the API name.